src/Nox.Reference.Data.Machine/DataSeeds/MachineDataSeeder.cs

```diff
@@ -1,42 +1,59 @@
 ﻿using System.Globalization;
-using System.Text.Json;
+using AutoMapper;
 using CsvHelper;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
-using Nox.Reference.Abstractions.MacAddresses;
+using Nox.Reference.Abstractions;
 using Nox.Reference.Common;
+using Nox.Reference.Data.Common;
 
-namespace Nox.Reference.GetData.CliCommands;
+namespace Nox.Reference.Data.Machine;
 
-public class MacAddressDataExtractCommand : ICliCommand
+internal class MachineDataSeeder : INoxReferenceDataSeeder
 {
     private const string SourceFilePath = @"MacAddresses\mac-vendor.csv";
-    private const string OutputFilePath = "Nox.Reference.MacAddresses.json";
 
     private readonly IConfiguration _configuration;
-    private readonly ILogger<MacAddressDataExtractCommand> _logger;
+    private readonly IMapper _mapper;
+    private readonly MachineDbContext _dbContext;
 
-    public MacAddressDataExtractCommand(
+    private readonly ILogger<MachineDataSeeder> _logger;
+
+    public MachineDataSeeder(
         IConfiguration configuration,
-        ILogger<MacAddressDataExtractCommand> logger)
+        IMapper mapper,
+        MachineDbContext dbContext,
+        ILogger<MachineDataSeeder> logger
+       )
     {
         _configuration = configuration;
+        _mapper = mapper;
+        _dbContext = dbContext;
         _logger = logger;
     }
 
-    public void Execute()
+    public void Seed()
     {
+        var dataSet = _dbContext
+            .Set<MacAddress>();
+
+        if (dataSet.Any())
+        {
+            return;
+        }
+
         _logger.LogInformation("Getting MAC Address data...");
 
         var sourceOutputPath = _configuration.GetValue<string>(ConfigurationConstants.SourceDataPathSettingName)!;
-        var targetOutputPath = _configuration.GetValue<string>(ConfigurationConstants.TargetDataPathSettingName)!;
 
         var sourceFilePath = Path.Combine(sourceOutputPath, SourceFilePath);
 
         if (!File.Exists(sourceFilePath))
         {
             DownloadSourceFileAsync()
```

Comment:
should we make all of this async?

Reply:
Why not, They can be performed in parallel.

Reply:
We may do it later, but why I'm not sure that it needs for now in console app.